### PR TITLE
[RW-3539][risk=no] New landing page for users with no workspaces

### PR DESCRIPTION
**Pull request by ericsong**
New
Screen Shot 2019-10-23 at 12 35 02 PM

Old
Screen Shot 2019-10-23 at 12 36 11 PM

### CODE REVIEW

File: ui/src/app/styles/colors.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ export default {
   success: '#8BC990',
   // Used for tables
   light: '#E9ECEF',
+  lightBackground: '#E6E4Ed',
```
Comment (andy7i) [inline ui/src/app/styles/colors.ts]: Is this in our official list of colors? If not, we should ask Lou to add it in. Or maybe `light` is the color Lou wanted?

File: ui/src/app/pages/homepage/homepage.tsx [hunk 2]
```diff
@@ -390,24 +402,47 @@ export const Homepage = withUserProfile()(class extends React.Component<
...
-                              <SmallHeader>Recently Accessed Items</SmallHeader>
-                              <RecentResources/>
+                              {this.state.userHasWorkspaces ? (<React.Fragment>
+                                <SmallHeader>Recently Accessed Items</SmallHeader>
+                                <RecentResources/>
+                              </React.Fragment>) : <div style={{backgroundColor: colors.lightBackground,
+                                color: colors.primary,
+                                borderRadius: 10,
+                                padding: '10px 30px',
+                                margin: '30px 0px'}}>
```
Comment (andy7i) [inline ui/src/app/pages/homepage/homepage.tsx]: Looks like the sides of the fadebox are drawn over

Comment (andy7i) [inline ui/src/app/pages/homepage/homepage.tsx]: Recommend using rem instead of px. 

further info
https://engageinteractive.co.uk/blog/em-vs-rem-vs-px

File: ui/src/app/pages/homepage/recent-workspaces.tsx [hunk 3]
```diff
@@ -42,22 +41,30 @@ export const RecentWorkspaces = withUserProfile()
...
-        })
-      }
-      {this.state.loading && <SpinnerOverlay dark={true} />}
-    </FlexRow>;
+    return <React.Fragment>
+        {!this.state.loading && this.state.recentWorkspaces.length === 0 ? (
+          <div style={{color: colors.primary, padding: '10px 30px'}}>
+            <h2 style={{fontWeight: 600}}>Create your first Workspace</h2>
+            <div>As you create your workspaces, this area will store your most recent workspaces.</div>
+            <div>To see all workspaces created, click on <b>See all Workspaces</b> to the right.</div>
```
Comment (andy7i) [inline ui/src/app/pages/homepage/recent-workspaces.tsx]: The mock shows the `See all Workspaces` link greyed out, but the code leaves it enabled. *However* I think it's okay to leave it enabled. It might be worth asking Lou though.

File: ui/src/app/pages/homepage/homepage.tsx [hunk 4]
```diff
@@ -390,24 +402,47 @@ export const Homepage = withUserProfile()(class extends React.Component<
...
+                                <h2 style={{fontWeight: 600, marginTop: 0}}>Here are some tips to get you started:</h2>
+                                <CustomBulletList>
+                                  <CustomBulletListItem bullet='→'>
+                                    Create a <a href='https://support.google.com/chrome/answer/2364824'>Chrome Profile </a>
```
Comment (andy7i) [inline ui/src/app/pages/homepage/homepage.tsx]: The text looks a bit small compared to the mock, but maybe that's just the screenshot. The mocks have the text at 14px, which I think is the smallest we go.
This also applies to the text:
```
As you create your workspaces, this area will store your most recent workspaces. 
To see all workspaces created, click on See all Workspaces to the right.
```

File: ui/src/app/pages/homepage/homepage.tsx [hunk 5]
```diff
@@ -390,24 +402,47 @@ export const Homepage = withUserProfile()(class extends React.Component<
                               <RecentWorkspaces />
                             </FlexColumn>
                             <FlexColumn>
-                              <SmallHeader>Recently Accessed Items</SmallHeader>
-                              <RecentResources/>
+                              {this.state.userHasWorkspaces ? (<React.Fragment>
+                                <SmallHeader>Recently Accessed Items</SmallHeader>
+                                <RecentResources/>
+                              </React.Fragment>) : <div style={{backgroundColor: colors.lightBackground,
```
Comment (andy7i) [inline ui/src/app/pages/homepage/homepage.tsx]: nit: for readability, could we format the ternary a bit differently? Something like:

```
{this.state.userHasWorkspaces ?

(<React.Fragment><SmallHeader>Recently Accessed Items</SmallHeader>
     <RecentResources/></React.Fragment>) : 

<div style={{backgroundColor: colors.lightBackground ..........
```

formatting multi-line ternary https://eslint.org/docs/rules/multiline-ternary

File: ui/src/app/pages/homepage/homepage.tsx [hunk 6]
```diff
@@ -390,24 +402,47 @@ export const Homepage = withUserProfile()(class extends React.Component<
...
+                                margin: '30px 0px'}}>
+                                <h2 style={{fontWeight: 600, marginTop: 0}}>Here are some tips to get you started:</h2>
+                                <CustomBulletList>
+                                  <CustomBulletListItem bullet='→'>
+                                    Create a <a href='https://support.google.com/chrome/answer/2364824'>Chrome Profile </a>
+                                    with your All of Us Researcher workbench Google account. This will keep your Workbench
+                                    browser sessions isolated from your other Google accounts.
+                                  </CustomBulletListItem>
+                                  <CustomBulletListItem bullet='→'>
+                                    Check out <a onClick={() => navigate(['library'])}> Featured Workspaces </a>
```
Comment (andy7i) [inline ui/src/app/pages/homepage/homepage.tsx]: not your code or within the scope of this PR -- this link doesn't play nicely with command+click to open in a new tab -- something we need to refactor in the app.

File: ui/src/app/pages/homepage/recent-workspaces.tsx [hunk 7]
```diff
@@ -42,22 +41,30 @@ export const RecentWorkspaces = withUserProfile()
...
+          </div>) : (<div>
```
Comment (andy7i) [inline ui/src/app/pages/homepage/recent-workspaces.tsx]: same comment about multi-line ternary formatting as before

File: ui/src/app/pages/homepage/recent-workspaces.tsx [hunk 8]
```diff
@@ -42,22 +41,30 @@ export const RecentWorkspaces = withUserProfile()
...
+              })
+            }
+          </FlexRow>
+        </div>)
+        }
+    </React.Fragment>;
   }
 });
+
+//       {this.state.loading && <SpinnerOverlay dark={true} />}
```
Comment (andy7i) [inline ui/src/app/pages/homepage/recent-workspaces.tsx]: did you mean to leave this as a comment? / any reason for removing this?

File: ui/src/app/pages/homepage/recent-workspaces.tsx [hunk 9]
```diff
@@ -42,22 +41,30 @@ export const RecentWorkspaces = withUserProfile()
...
-            accessLevel={recentWorkspace.accessLevel}
-            reload={() => this.loadWorkspaces()}
-          />;
-        })
-      }
-      {this.state.loading && <SpinnerOverlay dark={true} />}
-    </FlexRow>;
+    return <React.Fragment>
+        {!this.state.loading && this.state.recentWorkspaces.length === 0 ? (
+          <div style={{color: colors.primary, padding: '10px 30px'}}>
```
Comment (andy7i) [inline ui/src/app/pages/homepage/recent-workspaces.tsx]: maybe this isn't the best place for putting the padding? We can add padding in the parent div of RecentWorkspaces so that the parent is aware of how much padding it needs.

File: ui/src/app/components/headers.tsx [hunk 10]
```diff
@@ -19,6 +19,9 @@ export const styles = {
     fontWeight: 800,
     fontSize: '1rem',
   },
+  lessBold: {
```
[the remark this answers is not in the document]
Comment (ericsong) [reply ui/src/app/components/headers.tsx]: Do I need to be invited to this? The link isn't working for me.

File: ui/src/app/pages/homepage/homepage.tsx (hunk 6 above)
Comment (ericsong) [reply ui/src/app/pages/homepage/homepage.tsx]: yeah... this is what prompted my discussion at the eng-sync yesterday.

File: ui/src/app/pages/homepage/recent-workspaces.tsx (hunk 8 above)
Comment (ericsong) [reply ui/src/app/pages/homepage/recent-workspaces.tsx]: forgot! Adding it back

File: ui/src/app/components/headers.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (calbach) [reply ui/src/app/components/headers.tsx]: ^ same. If we actually had an AoU style guide this entire time that several team members did not have access to I'll have some follow-up questions...

File: ui/src/app/pages/homepage/recent-workspaces.tsx (hunk 9 above)
Comment (ericsong) [reply ui/src/app/pages/homepage/recent-workspaces.tsx]: The parent div won't work in this case unless I add a new div in between. The parent div contains the Workspace header which does not need the padding.

Is there an anti pattern with padding your own content?

File: ui/src/app/components/CustomBulletList.tsx [hunk 11]
```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
+
+export const CustomBulletList = ({children}) => {
+  return <ul style={{position: 'relative', listStyle: 'none', marginLeft: 0, paddingLeft: '1.2em'}}>{children}</ul>;
+};
+
+export const CustomBulletListItem = ({bullet, children}) => {
+  return <React.Fragment>
+    <div style={{position: 'absolute', left: -2}}> {bullet} </div> <li> {children} </li>
```
[the remark this answers is not in the document]
Comment (ericsong) [reply ui/src/app/components/CustomBulletList.tsx]: I had trouble getting it to work with flexbox. My guess is that flexbox can be awkward in "
hacky" css tricks like this.

File: ui/src/app/pages/homepage/homepage.tsx (hunk 6 above)
Comment (calbach) [reply ui/src/app/pages/homepage/homepage.tsx]: Note: here is the react conversion context I was mentioning: https://github.com/all-of-us/workbench/pull/1893#r265816213

File: ui/src/app/pages/homepage/homepage.tsx (hunk 2 above)
Comment (ericsong) [reply ui/src/app/pages/homepage/homepage.tsx]: What do you mean? I don't think I'm seeing this.

File: ui/src/app/components/headers.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (andy7i) [reply ui/src/app/components/headers.tsx]: sent!

File: ui/src/app/pages/homepage/recent-workspaces.tsx (hunk 9 above)
Comment (andy7i) [reply ui/src/app/pages/homepage/recent-workspaces.tsx]: Yeah I think you still want margin here. I'm realizing it probably doesn't need to be in a parent element.

More info: https://stackoverflow.com/questions/2189452/when-to-use-margin-vs-padding-in-css

File: ui/src/app/pages/homepage/recent-workspaces.tsx [hunk 12]
```diff
@@ -42,29 +43,34 @@ export const RecentWorkspaces = withUserProfile()
...
-          </FlexRow>
-        </div>)
-        }
+      {
+        this.state.loading ?
+          <SpinnerOverlay dark={true} /> :
+
+          this.state.recentWorkspaces.length === 0 ?
+
+            <div style={{color: colors.primary, padding: '.5em 2em'}}>
```
Comment (andy7i) [inline ui/src/app/pages/homepage/recent-workspaces.tsx]: tl;dr from https://stackoverflow.com/questions/2189452/when-to-use-margin-vs-padding-in-css

>TL;DR: By default I use margin everywhere, except when I have a border or background and want to increase the space inside that visible box.

I think we should use margins here.

Comment (ericsong) [reply ui/src/app/pages/homepage/recent-workspaces.tsx]: gotcha. will do

**Comment by ericsong** [comment]
poof. no more padding @andy7i
